docs/Design/Null Checking.md

```diff
@@ -0,0 +1,148 @@
+# Null Checking
+
+Creates checks for if the passed variable has value null. 
+
+A null variable is one that has no other value stored in it.
+
+Can be used to check whether the value is or is not null. 
+
+## Commands
+
+### `is null`
+
+`is null : variable`
+
+Creates a test that will return true if the variable is `null`.
+
+### `not null`
+
+`not null : variable`
+
+Creates a test that will return false if the variable is `null`.
+
+## Usage
+
+```
+is null : foo
+```
+
+```
+not null : bar
+```
+
+### CSharp
+
+```csharp
+foo == null
+```
+
+```csharp
+bar != null
+```
+
+### Java
+
+```java
+foo == null
+```
+
+```java
+bar != null
+```
+
+### Python
+
+```python
+foo is None
+```
+
+```python
+bar is not None
+```
+
+### Ruby
+
+```ruby
+foo.nil?
+```
+
+```ruby
+!bar.nil?
+```
+
+### TypeScript
+
+```typescript
+typeof foo === undefined

```

Comment:
This isn't quite how you check undefined-ness in TypeScript. Since you should be getting familiar with how to do this command in all languages (or at least enough to proofread these little pieces of code), I leave it up to you to find the correct way.

Hint: there are two preferred ones for `undefined`, and I prefer using the `typeof` operator. Bonus points if you find out why!